modules/xmpp/SdpConsistency.js

```diff
@@ -0,0 +1,176 @@
+import * as transform from 'sdp-transform';
+
+/**
+ * Begin helper functions
+ */
+/**
+ * Given a video mline (as parsed from transform.parse),
+ *  return the single primary video ssrcs
+ * @param {object} videoMLine the video MLine from which to extract the
+ *  primary video ssrc
+ * @returns {number} the primary video ssrc
+ */
+function getPrimarySsrc (videoMLine) {
+    if (!videoMLine.ssrcs) {
+        return;
+    }
+    let numSsrcs = videoMLine.ssrcs
+        .filter(ssrcInfo => ssrcInfo.attribute === "msid")
+        .map(ssrcInfo => ssrcInfo.id)
+        .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
```

Comment:
ah....oops...yes i see i am double filtering here...i think i forgot to remove the msid filtering line before (i think it's cleaner not to assume there is an msid line necessarily).  i'll clean this up.